kytos/core/tag_ranges.py

```diff
@@ -105,23 +106,35 @@ def range_intersection(
             get_validated_tags() for also list[int]
     """
     a_i, b_i = 0, 0
-    while a_i < len(ranges_a) and b_i < len(ranges_b):
+    common_diff = 1
+    if reverse:
+        a_i = len(ranges_a) - 1
+        b_i = len(ranges_b) - 1
+        common_diff = -1
+    while 0 <= a_i < len(ranges_a) and 0 <= b_i < len(ranges_b):
         fst_a, snd_a = ranges_a[a_i]
         fst_b, snd_b = ranges_b[b_i]
         # Moving forward with non-intersection
         if snd_a < fst_b:
-            a_i += 1
+            if not reverse:
+                a_i += common_diff
+            else:
+                b_i += common_diff
         elif snd_b < fst_a:
-            b_i += 1
+            if not reverse:
+                b_i += common_diff
+            else:
+                a_i += common_diff
         else:
             # Intersection
             intersection_start = max(fst_a, fst_b)
             intersection_end = min(snd_a, snd_b)
             yield [intersection_start, intersection_end]
-            if snd_a < snd_b:
-                a_i += 1
+            if_statement = snd_a < snd_b if not reverse else fst_a > fst_b
```

Comment:
`if_statement` var needs a better name, try to find something more meaningful.

Reply:
Change the variable name to `move_from_a`.